src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java

```diff
@@ -171,33 +181,36 @@ public void successiveInactiveEdition() {
 		}
 	}
 
-	@And("User clicks delete condition")
-	public void delete() {
-		String name = conditionsPage.getFirstConditionName();
-		if (StringUtils.isNotBlank(name)) {
+	@And("User clicks delete first active condition")
+	public void deleteFirstActiveCondition() {
+		if (StringUtils.isNotBlank(conditionsPage.getFirstConditionName())) {
 			conditionsPage.deleteFirstActive();
+			conditionsPage.confirmDeleteCondition();
 		}
-		driver.findElement(By.cssSelector(".confirm")).click();
+	}
 
-		conditionsPage.clickInActiveTab();
-		name = conditionsPage.getFirstConditionName();
-		if (StringUtils.isNotBlank(name)) {
+   @And("User clicks delete first inactive condition")
+   public void deleteFirstInactiveCondition() {
+	   conditionsPage.clickInActiveTab();
+		if (StringUtils.isNotBlank(conditionsPage.getFirstConditionName())) {
 			conditionsPage.deleteFirstInActive();
+			conditionsPage.confirmDeleteCondition();
 		}
-		driver.findElement(By.cssSelector(".confirm")).click();
 	}
 
-	@Then("System should trash first condition")
-	public void SuccessfulDeletion() {
+	@Then("System should trash first active condition")
+	public void systemDeletesActiveCondition() {
 		String name = conditionsPage.getFirstConditionName();
 		if (StringUtils.isNotBlank(name)) {
-			assertNull(driver.findElement(By.linkText(name)));
+			assertNull(driver.findElement(By.xpath(name)));
 		}
-
-		conditionsPage.clickInActiveTab();
-		name = conditionsPage.getFirstConditionName();
+	}
+		
+	@Then("System should trash first inactive condition")
+		public void systemDeletesInActiveCondition() {
+		String name = conditionsPage.getFirstConditionName();
 		if (StringUtils.isNotBlank(name)) {
-			assertNull(driver.findElement(By.linkText(name)));
+			assertNull(driver.findElement(By.xpath(name)));
```

Comment:
isnt there anything specific than asserting an element.

Reply:
not been able to find one. would you care to suggest?

Reply:
@sherrif10 do you have any response on the caption [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/230/files#r751346120)?

Reply:
@kdaud  am okay with that approach after inspecting it.